packages/flutter/lib/src/widgets/widget_inspector.dart

```diff
@@ -1089,9 +1089,19 @@ mixin WidgetInspectorService {
       name: 'getRootWidgetSummaryTree',
       callback: _getRootWidgetSummaryTree,
     );
-    _registerServiceExtensionWithArg(
+    registerServiceExtension(
       name: 'getDetailsSubtree',
-      callback: _getDetailsSubtree,
+      callback: (Map<String, String> parameters) async {
+        assert(parameters.containsKey('objectGroup'));
+        final int subtreeDepth = int.parse(parameters['subtreeDepth'] ?? '2');
```

Comment:
passing in '2' just to parse it seems a bit ugly. How about this:
```dart
final String subtreeDepth = parameters['subtreeDepth'];
return <String, Object>{
          'result': _getDetailsSubtree(
            parameters['arg'],
            parameters['objectGroup'],
            subtreeDepth != null ? int.parse(subtreeDepth) : 2,
          ),
        };
```

There is a dart:sdk bug to make int.tryParse accept null which would let you write
int.tryParse(parameters['subtreeDepth']) ?? 2